### DISCUSSION TITLE
MDFr as rust crate

### DISCUSSION THREAD

**Discussion by jejdouay** [ratal/mdfr#9] [General]
I have been enjoying both privately the mdfreader for python for a few years and now I am moving to rust. I was delighted to see that you already have implemented something in rust ( awesome !)

However the code is for integration in python using pyO3. do you have any plans to get a crate for mdfr without the pyO3 ?

If you need help, tell me 

Thank you for your great work !

Jerome

**Comment by ratal** [comment]
Hi,
You can already use mdfr without pyO3, there is simple C API existing. But indeed you have the dependency in cargo.toml with numpy or python polars, it could be architectured in such a way that it becomes an optional. Do you have a concrete use case in mind ? What would you need to use mdfr without pyO3 ? You can most likely use polars for instance, all in pure Rust, or datafusion. I did not experiment it, should be possible and relatively easy.
For info, I am doing again a big refactoring, replacing arrow2 which is not maintained by arrow-rs, for version 0.6

**Comment by jejdouay** [comment]
Hi ,

I have been using the python version for a few years to extract data from
measurement made with INCA and run analysis and data extraction. However I
have the issue it is just to slow to handle the large data I have,
especially trying to simulate a function using the measured data as it runs
point per point.

I am also moving to rust and I  am moving my script and tools to rust.
However I need a crate / lib to be able to get the data from the MDF file
and get them in a form like pandas in Python. I am also moving to polars so
your code seems to be perfect.

For the moment I am creating a personal crate based on your code to be a
pure MDF reader in python without pyO3 and other. I will translate my
python scripts to crates in rust to get the data from mdf reader and run
data extraction, filtering, triggers and others to make analysis.

thank you for the tip about arrow2, I did not know. I will keep an eye on
your development.

thank you again for the great both in python and rust, makes my life
definitely easier !

Jerome
